api/mfx_dispatch/windows/include/mfx_vector.h

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2013-2019 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+#include "mfxstructures.h"
+#include <exception>
+
+namespace MFX 
+{
+    template <class T>
+    class iterator_tmpl 
+    {
+        template <class U> friend class MFXVector;
+        mfxU32 mIndex;
+        T* mRecords;
+        iterator_tmpl(mfxU32 index , T * records) 
+            : mIndex (index)
+            , mRecords(records)
+        {}
+    public:
+        iterator_tmpl() 
+            : mIndex ()
+            , mRecords() 
+        {}
+        bool  operator ==(const iterator_tmpl<T> & that )const 
+        {
+            return mIndex == that.mIndex;
+        }
+        bool  operator !=(const iterator_tmpl<T> & that )const 
+        {
+            return mIndex != that.mIndex;
+        }
+        mfxU32 operator - (const iterator_tmpl<T> &that) const 
+        {
+            return mIndex - that.mIndex;
+        }
+        iterator_tmpl<T> & operator ++() 
+        {
+            mIndex++;
+            return * this;
+        }
+        iterator_tmpl<T> & operator ++(int) 
+        {
+            mIndex++;
+            return * this;
+        }
+        T & operator *() 
+        {
+            return mRecords[mIndex];
+        }
+        T * operator ->() 
+        {
+            return mRecords + mIndex;
+        }
+    };
+
+    class MFXVectorRangeError : public std::exception
+    {
+    };
+
+    template <class T>
+    class MFXVector  
```

Comment:
can we get rid of this?

Reply:
No :(

Reply:
@gkrivor : why?

Reply:
As we recommend embedding dispatcher into user app on Windows, we should minimize dependencies. Using std::vector would bring additional dependency from runtime libs. We can consider _optional_ standard implementation (define?) but I'm not sure if it worth it.

Reply:
so, we avoid std::vector and freely use std::exception just few lines above. I am not sure this makes sense to me. Wise versa can make some sense. Native-C dispatcher also might make sense. This - does not.

Reply:
We also use  `std::string DispatcherLog_GetMFXImplString(int impl)` Forbidding `std::vector` from that perspective looks inconsistent.